pkg/collector/corechecks/oracle/oracle_t.go

```diff
@@ -0,0 +1,313 @@
+	conn, err := sql.Open("oracle", databaseUrl)
+	assert.NoError(t, err)
+
+	err = conn.Ping()
+	assert.NoError(t, err)
+
+}
+
+func TestConnection(t *testing.T) {
+	databaseUrl := fmt.Sprintf(`user="%s" password="%s" connectString="%s"`, USER, PASSWORD, TNS_ALIAS)
+	db, err := sqlx.Open("godror", databaseUrl)
+	assert.NoError(t, err)
+	err = db.Ping()
+	assert.NoError(t, err)
+
+	databaseUrl = fmt.Sprintf(`user="%s" password="%s" connectString="%s:%d/%s"`, USER, PASSWORD, HOST, PORT, SERVICE_NAME)
+	_, err = sqlx.Open("godror", databaseUrl)
+	assert.NoError(t, err)
+	err = db.Ping()
+	assert.NoError(t, err)
+
+}
+
+func demuxOpts() aggregator.AgentDemultiplexerOptions {
+	opts := aggregator.DefaultAgentDemultiplexerOptions(nil)
+	opts.FlushInterval = 1 * time.Hour
+	opts.DontStartForwarders = true
+	return opts
+}
+
+func connectToDB(driver string) (*sqlx.DB, error) {
+	var connStr string
+	if driver == "godror" {
+		connStr = fmt.Sprintf(`user="%s" password="%s" connectString="%s:%d/%s"`, USER, PASSWORD, HOST, PORT, SERVICE_NAME)
+	} else if driver == "oracle" {
+		connStr = go_ora.BuildUrl(HOST, PORT, SERVICE_NAME, USER, PASSWORD, map[string]string{})
+	} else {
+		return nil, fmt.Errorf("wrong driver: %s", driver)
+	}
+
+	db, err := sqlx.Open(driver, connStr)
+	if err != nil {
+		return nil, fmt.Errorf("failed to connect to oracle instance: %w", err)
+	}
+	err = db.Ping()
+	if err != nil {
+		return nil, fmt.Errorf("failed to ping oracle instance: %w", err)
+	}
+	return db, nil
+}
+
+func TestChkRun(t *testing.T) {
+	aggregator.InitAndStartAgentDemultiplexer(demuxOpts(), "")
+
+	chk.dbmEnabled = true
+	chk.config.QueryMetrics = true
+
+	chk.config.InstanceConfig.InstantClient = false
+
+	type RowsStruct struct {
+		N                      int    `db:"N"`
+		SQLText                string `db:"SQL_TEXT"`
+		ForceMatchingSignature uint64 `db:"FORCE_MATCHING_SIGNATURE"`
+	}
+	r := RowsStruct{}
+
+	for _, tnsAlias := range []string{"", TNS_ALIAS} {
+		chk.db = nil
+
+		chk.config.InstanceConfig.TnsAlias = tnsAlias
+		var driver string
+		if tnsAlias == "" {
+			driver = common.GoOra
+			chk.config.InstanceConfig.InstantClient = false
+		} else {
+			driver = common.Godror
+		}
+
+		err := chk.Run()
+		assert.NoError(t, err, "check run with %s driver", driver)
+
+		err = chk.db.Get(&r, "select /* DDTEST */ 1 n from dual")
+		assert.NoError(t, err, "running test statement with %s driver", driver)
+		//m := make(map[uint64]int)
+		m := make(map[string]int)
+		m["2267897546238586672"] = 1
+		chk.statementsFilter = StatementsFilter{ForceMatchingSignatures: m}
+
+		statementMetrics, err := GetStatementsMetricsForKeys(&chk, "force_matching_signature", "AND force_matching_signature != 0", chk.statementsFilter.ForceMatchingSignatures)
+		assert.NoError(t, err, "running GetStatementsMetricsForKeys with %s driver", driver)
+		assert.Equal(t, 1, len(statementMetrics), "test query metrics captured with %s driver", driver)
```

Comment:
assert has a `Len` method for this with a nice error message.

Reply:
I implemented the check with `assert.Len`.